File: FreakFightsFan.Shared/Features/Fights/Helpers/FightResult.cs
using System.Globalization;
using System.Resources;

namespace FreakFightsFan.Shared.Features.Fights.Helpers;

public enum FightResult
{
    Upcoming = 0,
    Win = 1,
    Loss = 2,
    Draw = 3,
    NoContest = 4,
}

public static class FightResultExtensions
{
    private const string _resourceFileNamespace = "FreakFightsFan.Shared.Localization.Resources.FightResultConsts";
    private static readonly ResourceManager _resourceManager = new(_resourceFileNamespace, typeof(FightResultExtensions).Assembly);
        
    public static string ToText(this FightResult fightResult)
    {
        return fightResult switch
        {
            FightResult.Upcoming => _resourceManager.GetString("Upcoming", CultureInfo.CurrentCulture),
            FightResult.Win => _resourceManager.GetString("Win", CultureInfo.CurrentCulture),
            FightResult.Loss => _resourceManager.GetString("Loss", CultureInfo.CurrentCulture),
            FightResult.Draw => _resourceManager.GetString("Draw", CultureInfo.CurrentCulture),
            FightResult.NoContest => _resourceManager.GetString("NoContest", CultureInfo.CurrentCulture),
            _ => _resourceManager.GetString("Upcoming", CultureInfo.CurrentCulture),
        };
    }
}

File: freak-fights-fan/shared/features/fights/helpers/fightResult.ts
export enum FightResult {
    UPCOMING = 0,
    WIN = 1,
    LOSS = 2,
    DRAW = 3,
    NO_CONTEST = 4,
}

class FightResultExtensions {
    static toText(fightResult: FightResult): string {
        switch (fightResult) {
            case FightResult.UPCOMING:
                return 'Upcoming';
            case FightResult.WIN:
                return 'Win';
            case FightResult.LOSS:
                return 'Loss';
            case FightResult.DRAW:
                return 'Draw';
            case FightResult.NO_CONTEST:
                return 'No Contest';
            default:
                return 'Upcoming';
        }
    }
}
